### ISSUE TITLE
@Nullable annotation and Proguard

### ISSUE THREAD

**Issue by AndroidBunny** [square/moshi#734]
We implemented custom adapters to avoid null values as following:

```
private class DefaultStringAdapter {
    @FromJson
    fun fromJson(@Nullable jsonString: String?) = jsonString ?: ""
    @ToJson
    fun toJson(jsonString: String?) = jsonString
} 
```

Everything works like expected, but unfortunately it looks like when proguarding the annotation gets removed and therefore the adapter isn't used, what leads to an 

com.d.a.j: Non-null value 'msStation' was null at $.list[0].header.ms_station

Proguard rules defined as suggested in the README.

**Comment by NightlyNexus** [comment]
what is the Nullable annotation? does it have runtime retention?

**Comment by AndroidBunny** [comment]
It's javax.annotation.Nullable - RetentionPolicy.RUNTIME

**Comment by AndroidBunny** [comment]
Any ideas on how to make this work?

**Comment by NightlyNexus** [comment]
Can you make a reproducible sample?

**Comment by AndroidBunny** [comment]
Yes, I uploaded a very similar project here: https://github.com/AndroidBunny/MoshiAnnotation

You can see the difference by building DevDebug and DevProDebug. The first one runs how it should, the second one throws "com.c.a.j: Non-null value 'locationInfo' was null at $.list[1].header.location_info"

The Annotation is used in the MoshiModule.

**Comment by AndroidBunny** [comment]
Any updates on this one?

**Comment by ZacSweers** [comment]
Sorry for the delay! You need to add a proguard rule to keep that javax annotation, as proguard will remove it if it deems it as unused even if its retention policy is `RUNTIME`.

```proguard
-keep @interface javax.annotation.Nullable
```

**Comment by mdeandrea-mrmilu** [comment]
This still a issue? If not, please close
